### PR TITLE
Add support for ARM architectures

### PR DESCRIPTION
**Pull request by jngrad**
Description of changes:
- adjust test tolerances for architectures that don't have double extended precision, e.g. ARM
- add sanity checks in chain analysis functions and handle the corner case `number_of_chains=1`
- re-enable support for fast-math mode

### CODE REVIEW

**Comment by jngrad** [comment]
With these changes, the python branch passes the testsuite in fast-math mode (GitLab-CI [build 341966](https://gitlab.icp.uni-stuttgart.de/espressomd/espresso/-/jobs/341966)) and 4.2.1 passes the testsuite on ARM Neoverse-N1 (Koji [task 99951249](https://koji.fedoraproject.org/koji/taskinfo?taskID=99951249)).

File: src/core/unit_tests/thermostats_test.cpp [hunk 1]
```diff
@@ -42,7 +42,7 @@
 
 // multiply by 100 because BOOST_CHECK_CLOSE takes a percentage tolerance,
 // and by 6 to account for error accumulation in thermostat functions
-constexpr auto tol = 6 * 100 * std::numeric_limits<double>::epsilon();
+auto constexpr tol = 8. * 100. * std::numeric_limits<double>::epsilon();
```
Comment (reinaual) [inline src/core/unit_tests/thermostats_test.cpp]: This comment now doesn't make sense anymore, no?

**Review by reinaual — APPROVED** [review]
